### PR TITLE
RIP old networking layer

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/488551667048375/1203157703899144/f

### Description
Remove the legacy networking layer.

I have also removed:
* diagnostics screen from AppTP dev settings as it was more related to old networking layer
* transparency mode in AppTP dev settings as it only supported old networking layer

### Steps to test this PR

_Test fresh install_
- [x] fresh install from this branch
- [x] enable AppTP
- [x] verify it blocks trackers
- [x] verify all AppTP screens work as expected

_Test app update_
- [x] install from develop
- [x] enable AppTP
- [x] open apps to get tracking attempts
- [x] update app from this branch
- [x] verify no data has been lost
- [x] verify AppTP continues tracking
- [x] verify all AppTP screen work as expected

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2435** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/2435" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/2435?utm_source=stack-comment).

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/health/AppTPHealthMonitor.kt [hunk 1]
```diff
@@ -89,7 +72,6 @@ class AppTPHealthMonitor @Inject constructor(
 
         private val TUN_READ_ALERT_SAMPLES: Int = (4.minutes.inWholeMilliseconds / MONITORING_FREQUENCY_MS).toInt()
```
Comment (karlenDimla) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/health/AppTPHealthMonitor.kt:73]: NIT: not needed anymore

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/health/HealthClassifier.kt [hunk 2]
```diff
@@ -168,17 +58,6 @@ class HealthClassifier @Inject constructor(
...
-            numerator: Long,
-            denominator: Long
-        ): Double {
-            if (denominator == 0L) return 0.0
-            return numerator.toDouble() / denominator * 100
-        }
-    }
 }
 
 data class QueueReads(
```
Comment (karlenDimla) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/health/HealthClassifier.kt:63]: Do we still need this object?

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/stats/AppTrackerBlockingStatsRepository.kt [hunk 3]
```diff
@@ -106,15 +99,9 @@ class RealAppTrackerBlockingStatsRepository @Inject constructor(
 ) : AppTrackerBlockingStatsRepository {
```
Comment (karlenDimla) [inline app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/stats/AppTrackerBlockingStatsRepository.kt:99]: `data class DataTransfer(
    val dataSize: Long = 0,
    val numberPackets: Long = 0
)`

seems not needed anymore.

**Review by karlenDimla — COMMENTED** [review]
I tested changes and it all looks good.

I think there's a few diagnostics related strings in strings-vpn.xml that we don't need anymore.

**Review by karlenDimla — APPROVED** [review]

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/health/AppTPHealthMonitor.kt (hunk 1 above)
Comment (aitorvs) [reply app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/health/AppTPHealthMonitor.kt:73]: nice, will remove

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/health/HealthClassifier.kt (hunk 2 above)
Comment (aitorvs) [reply app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/health/HealthClassifier.kt:63]: nice, will remove `QueueReads` class

File: app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/stats/AppTrackerBlockingStatsRepository.kt (hunk 3 above)
Comment (aitorvs) [reply app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/stats/AppTrackerBlockingStatsRepository.kt:99]: Awesome, will remove

**Comment by aitorvs** [comment]
> I tested changes and it all looks good.
> 
> I think there's a few diagnostics related strings in strings-vpn.xml that we don't need anymore.

yeah, will clean up the strings a bit later. I didn't just yet because didn't want to cause conflicts on the smartling PR

**Pull request merged**